api/magic_link.go

```diff
@@ -50,19 +55,35 @@ func (a *API) MagicLink(w http.ResponseWriter, r *http.Request) error {
 			if err != nil {
 				internalServerError("error creating user").WithInternalError(err)
 			}
-			newBodyContent := `{"email":"` + params.Email + `","password":"` + password + `"}`
-			r.Body = ioutil.NopCloser(strings.NewReader(newBodyContent))
-			r.ContentLength = int64(len(newBodyContent))
+
+			signUpParams := &SignupParams{
+				Email:    params.Email,
+				Password: password,
+				Data:     params.Metadata,
+			}
+			newBodyContent, err := json.Marshal(signUpParams)
+			if err != nil {
+				return badRequestError("Could not parse metadata: %v", err)
+			}
+			r.Body = ioutil.NopCloser(strings.NewReader(string(newBodyContent)))
+			r.ContentLength = int64(len(string(newBodyContent)))
```

Comment:
Sad. This doesn't give the exact number of bytes. Additionally, it seems like we could potentially remove this line as per [this thread](https://stackoverflow.com/questions/49806673/how-to-know-content-length-of-my-request-body-in-order-to-add-it-in-my-request-h)